### DISCUSSION TITLE
update onDelete when updateSchema

### DISCUSSION THREAD

**Discussion by hubbusysyuhada** [mikro-orm/mikro-orm#3032] [Q&A]
I had an Entity that I've built months ago that stated something like this
```
@Entity({ tableName: "myTable" })
export default class myTable {
  @PrimaryKey()
  id: string = v4();

  @Property({ index: true, unique: true })
  name!: string;

  @OneToMany(() => OtherEntity, (OtherEntity) => OtherEntity.myTable)
  relation = new Collection<OtherEntity>(this);

  constructor(name: string) {
    this.name = name;
  }
}
```

then I initialize the Mikro-ORM instance and update it using `updateSchema()` method from `generator` instance mentioned on here https://mikro-orm.io/docs/schema-generator

now, I changed the foreign key column names from `relation` into `new_relation` with option `{onDelete: "CASCADE"}`. The foreign key column did changed, but the ON DELETE CASCADE has not been implemented.

Do you know how to fix it?

**Comment by B4nan** [comment]
1:m is virtual relation, you need to put this onto the owning side, which represents the FK - a real database column. `onDelete` should not even be possible in 1:m decorator options and should produce TS error.

**Reply by hubbusysyuhada** [reply]
my bad, I meant the entities were
```
  @ManyToOne(() => OtherEntity)
  relation!: OtherEntity;
```

and change to
```
  @ManyToOne(() => OtherEntity, {onDelete: "CASCADE"})
  new_relation!: OtherEntity;
```

the `new_relation` column's `ON DELETE` was still no action instead of CASCADE

**Reply by B4nan** [reply]
What version are you on?

**Reply by hubbusysyuhada** [reply]
4.5.9

**Reply by B4nan** [reply]
then upgrade to v5, this is not supported in v4

**Reply by hubbusysyuhada** [reply]
i see, okay will try. Thanks!
